### DISCUSSION TITLE
Implement an easily replaceable Printer for `console`

### DISCUSSION THREAD

**Discussion by Razican** [boa-dev/boa#1532] [Ideas]
**Current situation**
We currently have the [`Console`](https://console.spec.whatwg.org/) standard implemented, but we don't implement it with the common [`Logger`](https://console.spec.whatwg.org/#logger), [`Printer`](https://console.spec.whatwg.org/#printer) and [`Formatter`](https://console.spec.whatwg.org/#formatter) operations. This prevents crates using Boa from creating their own printers that will for example not print anything but save the result to a file or a database.

**Proposed feature**
We should implement an easily replaceable `Printer` that would either be passed to the Realm / Interpreter, so that by default, the `console` will work as it does now, but we should have a way of creating the `console` object to use a custom `Printer`.

**How do we do this?**
We should have a `printer` object that implements the `Printer` trait, which would be something like:
```rust
trait Printer {
    fn print(&mut self, logLevel: LogLevel, args: &[Value]);
}
```

Maybe the trait needs to be modified, and we would need a `LogLevel` enum, but I think the implementation should go in this direction. The idea behind having the logger use `&mut self` is because we might want a printer that stores the logs in memory, or in a buffer, or that uses `Write` to write to a log file.

What I'm not very sure is on what the `options` should be in the `Printer`. Maybe we can just skip that argument for now.

**Comment by twop** [comment]
Im wondering why this feature request didn't get more traction, that seems like that can be a common concern
